### PR TITLE
Properly handle new remote tracks

### PR DESCRIPTION
**Pull request by LVala**

### CODE REVIEW

File: test/ex_webrtc/peer_connection_test.exs [hunk 1]
```diff
@@ -37,6 +37,9 @@ defmodule ExWebRTC.PeerConnectionTest do
 
     assert_receive {:ex_webrtc, ^pc, {:track, %MediaStreamTrack{kind: :audio}}}
 
+    {:ok, answer} = PeerConnection.create_answer(pc)
```
Comment (mickel8) [inline test/ex_webrtc/peer_connection_test.exs:40]: Would be nice to add a test that should fail in the previous implementation but should pass in the current one

File: lib/ex_webrtc/peer_connection.ex [hunk 2]
```diff
@@ -619,19 +623,71 @@ defmodule ExWebRTC.PeerConnection do
...
-    Enum.reduce_while(sdp.media, {:ok, transceivers}, fn mline, {:ok, transceivers} ->
-      case ExSDP.Media.get_attribute(mline, :mid) do
-        {:mid, mid} ->
-          transceivers = RTPTransceiver.update_or_create(transceivers, mid, mline, config)
-          {:cont, {:ok, transceivers}}
+  defp update_transceivers(state, sdp) do
+    Enum.reduce(sdp.media, state.transceivers, fn mline, transceivers ->
+      {:mid, mid} = ExSDP.Media.get_attribute(mline, :mid)
+
+      # TODO: reusing transceivers
```
Comment (mickel8) [inline lib/ex_webrtc/peer_connection.ex:630]: Could we elaborate on this comment? I'm slowly getting lost

**Review by mickel8 — CHANGES_REQUESTED** [review]

File: lib/ex_webrtc/peer_connection.ex (hunk 2 above)
Comment (LVala) [reply lib/ex_webrtc/peer_connection.ex:630]: I meant "recycling transceivers", we don't do that at the moment.

File: lib/ex_webrtc/peer_connection.ex [hunk 3]
```diff
@@ -619,19 +623,71 @@ defmodule ExWebRTC.PeerConnection do
...
+      {:mid, mid} = ExSDP.Media.get_attribute(mline, :mid)
+      {idx, transceiver} = find_transceiver(transceivers, mid)
+
+      direction = SDPUtils.get_media_direction(mline)
+
+      direction =
+        case source do
+          :local -> direction
+          :remote -> reverse_direction(direction)
+        end
```
[the remark this answers is not in the document]
Comment (LVala) [reply lib/ex_webrtc/peer_connection.ex:669]: As we discussed offline, this implementation is actually true to the spec.

[the remark this answers is not in the document]
Comment (mickel8) [reply lib/ex_webrtc/peer_connection.ex:669]: For future reference, see this comment https://github.com/w3c/webrtc-pc/issues/2751#issuecomment-1851583208

**Review by mickel8 — APPROVED** [review]

**Pull request merged**